### PR TITLE
Pol 21

### PR DESCRIPTION
**Pull request by rlafleur**
Description
S3 bucket policy as per POL-21 in Jira

Issues Resolved
Not applicable

### Contribution Check List

- [x] All tests pass.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] New functionality has been documented in CHANGELOG.MD

### CODE REVIEW

File: cost/aws/s3_bucket_size/aws_bucket_size.pt [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+      "bucket_name": buckets[i]["bucket_name"],
+      "creation_date": buckets[i]["creation_date"],
+      "region": buckets[i]["region"],
+      "host": "s3-" + buckets[i]["region"] + ".amazonaws.com",
+      "auth": "s3_" + buckets[i]["region"].replace(/-/g,"_")
+    }
+    )
+  }
+};
+//console.log("buckets", results)
```
Comment (cdwilhelm) [inline cost/aws/s3_bucket_size/aws_bucket_size.pt]: remove unnecessary comments

File: cost/aws/s3_bucket_size/aws_bucket_size.pt [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+      'Dimensions.member.2.Value': 'StandardStorage', 
+      'StartTime': start_date,
+      'EndTime': end_date,
+      'Period': "2592000",
+      'Statistics.member.1': 'Maximum',
+      'Statistics.member.2': 'Average',
+      'Statistics.member.3': 'Minimum'
+    }
+  }  
+  //console.log("printing aws_bucket_sizes_request bucket_name", request)
```
Comment (cdwilhelm) [inline cost/aws/s3_bucket_size/aws_bucket_size.pt]: remove unnecessary comments

File: cost/aws/s3_bucket_size/aws_bucket_size.pt [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+    var i = parseInt(Math.floor(Math.log(bytes) / Math.log(1000)), 10)
+	// account for input larger than TB
+	if (i >= 4) {
+	  var size = 4
+	} else {
+	  var size = i
+	}
+	
+	// console.log("i is " + i);
+  // console.log("size is " + size);
```
Comment (cdwilhelm) [inline cost/aws/s3_bucket_size/aws_bucket_size.pt]: please remove

File: cost/aws/s3_bucket_size/aws_bucket_size.pt [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+	  // console.log("avg_size is " + avg_size)
+      results.push (
+        {
+	      "bucket_name": bucket_name,
+	      "average": avg_size
+	    }
+	  )
+    }
+  }
+  //console.log("filteredbuckets", results)
```
Comment (cdwilhelm) [inline cost/aws/s3_bucket_size/aws_bucket_size.pt]: please remove

File: cost/aws/s3_bucket_size/README.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+### Policy Actions
+
+The following policy actions are taken on any resources found to be out of compliance.
+
+- Send an email report
+
+### Required Permissions
+
+This policy requires permissions to access RightScale resources (clouds, volumes, deployments, placement groups and tags).  Before applying this policy add the following roles to the user applying the policy.  The roles should be applied to all Accounts where the policy will run or the Organization. For more information on modifying roles visit the [Governance Docs](https://docs.rightscale.com/cm/ref/user_roles.html)
```
Comment (cdwilhelm) [inline cost/aws/s3_bucket_size/README.md]: I didn't see any rightscale permissions needed.  You'll want to show what AWS permissions are needed.

**Comment by rlafleur** [comment]
I removed comments from the policy template. A couple of questions:


  1.  I approved an update from Erwin Gonzalez in the README.md --> https://github.com/flexera/policy_templates/pull/306/files. Will this cause a merge conflict if I update the root README.md?
  2.  Do you need any more details than the following for the AWS permission “This policy requires permissions to read S3 buckets and retrieve S3 bucket statistics from CloudWatch.”? I don’t have anything more specific than that.

Please let me know when the ruby build errors are resolved and I’ll commit and push the changes.

Thanks,

Robert Lafleur
Principal Solution Engineer
Mobile: 512-657-7283
RLafleur@flexera.com<mailto:RLafleur@flexera.com>
www.flexera.com

**Comment by cdwilhelm** [comment]
Robert

The readme won't conflict until it's merged into the master, when that happens you can merge into your branch.

Regarding the permissions, it's actually a little more than that.  This policy is actually using cloudwatch and s3.  Also your policy is using the CM AWS creds which isn't mentioned in the readme and should be. The other AWS policies have some examples, such as this one https://github.com/flexera/policy_templates/tree/master/cost/aws/instance_cloudwatch_utilization


Curt


[https://resources.flexera.com/web/outlook/images/buyer_logo.png]
Curt Wilhelm
Senior Program Manager

United States

CWilhelm@flexera.com<mailto:CWilhelm@flexera.com>
www.flexera.com<http://www.flexera.com>

Flexera acquires RISC Networks:
Breakthrough cloud migration analytics platform. [https://resources.flexera.com/web/outlook/images/flexera_monarch_logo.png] <https://www.flexera.com/about-us/press-center/flexera-acquires-risc-networks.html>


________________________________

**Pull request merged**